### ISSUE TITLE
Question about ObstacleClusters::GetLaneGraph()

### ISSUE THREAD

**Issue by wanggujin** [ApolloAuto/apollo#9364]
Apollo 5.0
[ObstacleClusters::GetLaneGraph](https://github.com/ApolloAuto/apollo/blob/949d2b0619e96e723714445233eecfed6c3dbfea/modules/prediction/container/obstacles/obstacle_clusters.cc#L61)

why only update start_s here ?
there is no need to update end_s and adc_s ?

**Comment by kechxu** [comment]
@wanggujin Thanks for your question. The end_s is updated in obstacle.cc

**Comment by wanggujin** [comment]
@kechxu Thank you. but i can't find that in obstacle.cc.  can you give me a link?

**Comment by wanggujin** [comment]
@kechxu thanks. 
but you don't understand me. let me give you a example. Consider the following scenario：
A lane, such as lane1 and it's length is 100m, and two obstacles on it. obstacle1 is located x=0 and obstacle2 is located x=60.
when first call [ObstacleClusters::GetLaneGraph](https://github.com/ApolloAuto/apollo/blob/949d2b0619e96e723714445233eecfed6c3dbfea/modules/prediction/container/obstacles/obstacle_clusters.cc#L43) for obstacle1, the lane_graphs_ map not include the key "lane1", so we construct it(assume we construct 50m, so the new LaneGraph's start_s is 0 and end_s is 50).  But when second call the function for obstacle2 that it's also on lane1  the key "lane1" has existed in lane_graphs_  map,  so we directly modify the existed LaneGraph's start_s from 0 to 60, then  return it. Then the question becomes that the LaneGraph we returned for obstacle2, its start_s is 60, but end_s is still 50, because you don't modify it. 
i think it is a mistake. do you?

meanwhile, i have same question about adc_s.

**Comment by kechxu** [comment]
@wanggujin Yes, you are right. It seems to be a mistake. Thanks for bringing it to our attention. We will look into the code and fix this issue.
